libs/user_extensions/src/TTAlpsEvent.cpp

```diff
@@ -140,7 +140,7 @@ bool TTAlpsEvent::IsGoodMuonFromALP(int genMuonIndex) {
   // loop over pythia copies of the muon
   if (mother->IsMuon()) {
     while(mother->IsMuon()) {
-      motherIndex = muon->GetMotherIndex();
+      motherIndex = mother->GetMotherIndex();
       if (motherIndex < 0) return false;
       mother = asNanoGenParticle(genParticles->at(motherIndex));
     }
```

Comment:
Can we make this a method of `NanoGenParticle`? I.e. implement `GetFirstCopy` that will return the first copy of a particle regardless of its PID?

Reply:
I don't think I understand what you mean with which method, but we need the event to get the complete collection of genparticles to change the mothers